oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/migration/version/VersionCopier.java

```diff
@@ -111,6 +125,72 @@ public boolean copyVersionHistory(String versionableUuid, Calendar minDate, bool
         return false;
     }
 
+    private boolean hasNoConflicts(String versionHistoryPath, String versionableUuid, boolean preserveOnTarget, NodeState sourceVersionHistory) {
+        // if preserveOnTarget is true then check no conflicts which means version history has moved forward only
+        if (preserveOnTarget) {
+            NodeBuilder targetVersionHistory = getVersionHistoryBuilder(targetVersionStorage, versionableUuid);
+            if (targetVersionHistory.exists()) {
+                VersionComparator versionComparator = new VersionComparator();
+
+                // version history id not equal
+                boolean conflictingVersionHistory = !targetVersionHistory.getString(JCR_UUID).equals(sourceVersionHistory.getString(JCR_UUID));
```

Comment:
Since `NodeBuilder.getString()` can return null, I would compare them using `Objects.equal`

Reply:
we are in the version history node path which will mandatorily contain jcr:uuid property but Objects.equal comparison is good.